bitcoin/src/blockdata/locktime/absolute.rs

```diff
@@ -305,20 +308,45 @@ impl LockTime {
         }
     }
 
+    /// Returns true if this [`absolute::LockTime`] is satisfied by `other` lock.
+    ///
+    /// This function is useful if you wish to check a lock time against various `other` locks e.g.,
+    /// filtering out locks which cannot be satisfied.
```

Comment:
Shouldn't this be filtering out locks that are guaranteed to be satisfied? Creating conflicting locks is impossible in Bitcoin.

Reply:
I believe the existing text is correct, but it is possible that we're interpreting "filtering out" differently..

If a locktime can't be satisfied then you want to eliminate any part of your script that depends on it, since it's unsatisfiable.

Reply:
How it's possible to create unsatisfiable lock time in Bitcoin? I've seen people explaining that Bitcoin doesn't allow that because it'd greatly complicate reorgs.

Reply:
Within a complex script you can have locktime checks that simply aren't executed and wouldn't have been valid if they were. (e.g. by using `BOOLOR`)

Reply:
AFAIR all lock time checking opcodes have verify semantics so you can't put `FALSE` on stack if the lock time is *not* satisfied.

What I mean is you could have a situation where there are two checks: whether height is above `x` and `x + n`. `x + n` implies `x` and thus `x` is not needed because it's guaranteed to be satisfied.

Reply:
You can if you wrap it in an `IF` `ENDIF`. We do this in Miniscript all the time *and* we [filter by age](https://docs.rs/miniscript/latest/miniscript/policy/semantic/enum.Policy.html#method.at_age) as described in the doccomment we're discussing.

Reply:
Can you write an example script? I still can't figure out how.

Reply:
```
<B> OP_CHECKSIG OP_NOTIF
  <C> OP_CHECKSIG OP_NOTIF
    <e803> OP_CHECKSEQUENCEVERIFY OP_VERIFY
  OP_ENDIF
OP_ENDIF
<A> OP_CHECKSIG
```

This is the default example on https://bitcoin.sipa.be/miniscript/ -- you can simplify it a bit and click "Compile" but if you make it too simple it won't bother wrapping the CSV/CLTV in `IF`/`ENDIF`.

Reply:
I don't understand, it has only one `OP_CHECKSEQUENCEVERIFY` so there is no other to compare to.

Reply:
When you are analyzing the transaction you would use the sequence number from the transaction (or from a hypothetical transaction you are thinking of creating) to compare it.

Reply:
Ah, that makes sense. Given how long it took me to get it, I think we should have a clearer comment. Maybe something like this:

> Can be used to check if `OP_CHECKSEQUENCEVERIFY` allows specific sequence. Example:
> ```rust
> txin.sequence.is_implied_by_lock(csv_param)
> ```
> Or remove the smaller value of two `OP_CHECKSEQUENCEVERIFY` operations within one branch of the script.

Reply:
The example given is not quite right, we have to convert sequence to a relative lock time before calling `is_implied_by`. I've updated the comments in both `absolute` and `relative` modules. Done as a separate patch since the docs are similar.